addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -397,20 +397,24 @@ def fill_stock_move_bom_line_id(cr):
         cr,
         """
         UPDATE stock_move sm_update
-        SET bom_line_id = mbl.bom_line_id
-        FROM stock_move sm
-            INNER JOIN mrp_production mp
+        SET bom_line_id = q.bom_line_id
+        FROM (
+            SELECT sm.id, mbl.bom_line_id
+            FROM stock_move sm
+            INNER JOIN mrp_production mp 
             ON sm.raw_material_production_id = mp.id
-            LEFT JOIN (
+            LEFT JOIN (                           
                 SELECT min(id) AS bom_line_id, bom_id, product_id
-                FROM mrp_bom_line
-                GROUP BY bom_id, product_id
-                ) mbl ON (
-                    mbl.bom_id = mp.bom_id
-                    AND mbl.product_id = sm.product_id)
-        WHERE sm_update.raw_material_production_id = mp.id
-            AND sm_update.id = sm.id
-            AND sm_update.bom_line_id IS NULL;
+                FROM mrp_bom_line          
+                GROUP BY bom_id, product_id                
+            ) mbl ON (                               
+                mbl.bom_id = mp.bom_id      
+                AND mbl.product_id = sm.product_id
+            )
+            WHERE sm.raw_material_production_id = mp.id 
```

Comment:
```suggestion
                FROM mrp_bom_line
                GROUP BY bom_id, product_id
            ) mbl ON (
                mbl.bom_id = mp.bom_id
                AND mbl.product_id = sm.product_id
            )
            WHERE sm.raw_material_production_id = mp.id
```
delete trailing whitespace